doc/api/tls.md

```diff
@@ -2,90 +2,145 @@
 
     Stability: 2 - Stable
 
-Use `require('tls')` to access this module.
+The `tls` module provides an implementation of the Transport Layer Security
+(TLS) and Secure Socket Layer (SSL) protocols that is built on top of OpenSSL.
+The module can be accessed using:
 
-The `tls` module uses OpenSSL to provide Transport Layer Security and/or
-Secure Socket Layer: encrypted stream communication.
+```js
+const tls = require('tls');
+```
+
+## TLS/SSL Concepts
 
-TLS/SSL is a public/private key infrastructure. Each client and each
-server must have a private key. A private key is created like this:
+The TLS/SSL is a public/private key infrastructure (PKI). Each client and
+server must have a *private key*.
+
+Private keys can be generated in multiple ways. The example below illustrates
+use of the OpenSSL command-line interface to generate a 2048-bit RSA private
+key:
 
 ```
 openssl genrsa -out ryans-key.pem 2048
 ```
 
-All servers and some clients need to have a certificate. Certificates are public
-keys signed by a Certificate Authority or self-signed. The first step to
-getting a certificate is to create a "Certificate Signing Request" (CSR)
-file. This is done with:
+With TLS/SSL, all servers (and some clients) must have a *certificate*.
+Certificates are *public keys* that correspond to a private key, and that are
+digitally signed either by a Certificate Authority or by the owner of the
+private-key (such certificates are referred to as "self-signed"). The first
```

Comment:
Also, this paragraph is only "common case" true.  It's e.g. possible for a certificate to have been signed by a self-signed CA.  I understand that you probably don't want to over-complicate the primer, though.